smart_open/smart_open_lib.py

```diff
@@ -582,6 +586,8 @@ def _compression_wrapper(file_obj, filename, mode):
     if _need_to_buffer(file_obj, mode, ext):
         warnings.warn('streaming gzip support unavailable, see %s' % _ISSUE_189_URL)
         file_obj = io.BytesIO(file_obj.read())
+    if ext in COMPRESSED_EXT and mode.endswith('+'):
+        raise ValueError('impossible to open in read+write+compressed mode')
```

Comment:
Nothing is impossible :) Let's use a more positive exception message.

```suggestion
        raise ValueError('transparent (de)compression currently unsupported for mode %r' % mode)
```

Reply:
I would remove "currently" to remove the false sense of hope. The way codecs are implemented is incompatible with rw, and nobody is going to rewrite them in Python core.